design-documents/graph-ql/coverage/negotiableQuotes.graphqls

```diff
@@ -1,205 +1,332 @@
+    ): AddNegotiableQuoteItemsOutput @doc(description: "Add 1 or more products to a Negotiable Quote")
 
-    setShippingMethodsOnNegotiableQuote(
-        cartId: ID! @doc(description: "Cart ID")
-        shippingMethods: [ShippingMethodInput]! @doc(description: "Input for shipping method")
-    ): SetShippingMethodsOnNegotiableQuoteOutput @doc(description:"Set negotiable quote shipping method")
+    # Covers first half of https://docs.magento.com/user-guide/customers/account-dashboard-quotes.html#cancel-a-quote-request
+    closeNegotiableQuotes(
+        input: CloseNegotiableQuotesInput!
+    ): CloseNegotiableQuotesOutput @doc(description: "Mark a Negotiable Quote as closed, leaving it visible in the storefront")
+
+    # Covers second half of https://docs.magento.com/user-guide/customers/account-dashboard-quotes.html#cancel-a-quote-request
+    deleteNegotiableQuotes(
+        input: DeleteNegotiableQuotesInput!
+    ): DeleteNegotiableQuotesOutput @doc(description: "Delete a Negotiable Quote, removing it from the display in the storefront")
+
+    # Covers "Select Existing Address" in https://docs.magento.com/user-guide/customers/account-dashboard-quotes-negotiate.html#shipping-information
+    # "New Address" flow is covered by Mutation.createCustomerAddress
+    setNegotiableQuoteShippingAddress(
+        input: SetNegotiableQuoteShippingAddressInput!
+    ): SetNegotiableQuoteShippingAddressOutput @doc(description: "Assign one of buyers' existing addresses to a Negotiable Quote")
+
+    # Pending decision on design of file upload (design doc still pending decisions)
+    # addNegotiableQuoteFiles(
+    #     input: AddNegotiableQuoteFilesInput!
+    # ): AddNegotiableQuoteFilesInput
+}
+
+type AddNegotiableQuoteItemsOutput {
+    quote: NegotiableQuote
+    # TODO: We'll probably want to add the same
+    #       errors that can happen when adding an item
+    #       to a normal cart object, but will also have
+    #       some additional errors (i.e can't add an item
+    #       when a negotiable quote is in some statuses)
+}
+
+input SetNegotiableQuoteShippingAddressInput {
+    quote_id: ID! @doc(description: "ID obtained from NegotiableQuote type")
+    customer_address_id: ID! @doc(
+        description: "ID obtained from the CustomerAddress type. A new address can be added using Mutation.createCustomerAddress"
+    )
 }
 
-type Customer {
-    negotiable_quote (
-        id: ID!): NegotiableQuote @doc(description: "Get negotiable quote of a customer")
+type SetNegotiableQuoteShippingAddressOutput {
+    quote: NegotiableQuote
+}
 
+input AddNegotiableQuoteItemsInput {
+    quote_id: ID! @doc(description: "ID from a NegotiableQuote object")
+    # Implementation Note: This *should* be compatible with the new, single
+    # add to cart mutation. https://github.com/magento/architecture/blob/master/design-documents/graph-ql/coverage/AddProductsToCart.graphqls
+    cart_items: [CartItemInput!]!
```

Comment:
Please confirm that it is currently possible to add new items to the negotiable quote after it was requested (created based on cart)

Reply:
This is mentioned in the code comment where this type is used, on `Query.addNegotiableQuoteItems`.

https://devdocs.magento.com/guides/v2.4/b2b/negotiable-update.html#add-a-new-quote-item-to-the-negotiable-quote

![image](https://user-images.githubusercontent.com/5233399/88207504-82db2180-cc15-11ea-9025-ff35384c3358.png)

Reply:
Double-checked with @mbrinton01 to be safe - he said the docs are accurate